src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPPollingSource.java

```diff
@@ -87,4 +109,60 @@ public void onTerminate() {
     endUsingFolder();
   }
 
+  @Override
+  protected Message[] getMessages(Folder openFolder) {
+    try {
+      IMAPEmailPredicateBuilder matcher = getPredicateBuilder().orElseGet(() -> new IMAPEmailPredicateBuilder());
+
+      SearchTerm answeredTerm = getSearchTerm(Flags.Flag.ANSWERED, matcher.getAnswered(), false);
+      SearchTerm deletedTerm = getSearchTerm(Flags.Flag.DELETED, matcher.getDeleted(), false);
+      SearchTerm recentTerm = getSearchTerm(Flags.Flag.RECENT, matcher.getRecent(), false);
+      SearchTerm seenTerm = getSearchTerm(Flags.Flag.SEEN, matcher.getSeen(), false);
+
+      AndTerm searchTerm = new AndTerm(answeredTerm, deletedTerm);
+      searchTerm = new AndTerm(searchTerm, recentTerm);
+      searchTerm = new AndTerm(searchTerm, seenTerm);
+
+      if (matcher.getReceivedSince() != null) {
+        Date receivedSinceDate = convertLocalDateTimeToDate(imapMatcher.getReceivedSince());
+        ReceivedDateTerm receivedDateTerm = new ReceivedDateTerm(GE, receivedSinceDate);
+        searchTerm = new AndTerm(searchTerm, receivedDateTerm);
+      }
+
+      if (matcher.getReceivedUntil() != null) {
+        Date receivedUntilDate = convertLocalDateTimeToDate(imapMatcher.getReceivedUntil());
+        ReceivedDateTerm receivedDateTerm = new ReceivedDateTerm(LE, receivedUntilDate);
+        searchTerm = new AndTerm(searchTerm, receivedDateTerm);
+      }
+
+      if (matcher.getSentSince() != null) {
+        Date sentSinceDate = convertLocalDateTimeToDate(imapMatcher.getSentSince());
+        SentDateTerm sentDateTerm = new SentDateTerm(GE, sentSinceDate);
+        searchTerm = new AndTerm(sentDateTerm, searchTerm);
+      }
+
+      if (matcher.getSentUntil() != null) {
+        Date sentUntilDate = convertLocalDateTimeToDate(imapMatcher.getSentUntil());
+        SentDateTerm sentDateTerm = new SentDateTerm(LE, sentUntilDate);
+        searchTerm = new AndTerm(searchTerm, sentDateTerm);
```

Comment:
In order to avoid duplicate processing should we use GE or G? (Same applies for LE or L)

Reply:
Based on the previous behaviour we had for these filters, I think it would be better to leave the inclusive range to avoid breaking backwards

Reply:
By duplicate processing do you mean sending the same email twice to the flow? I think this is about making the server filter email based on the matcher. Then you will still test re incoming results with the matcher, and the idempotency and watermarking features will aply.